netbeans.apache.org/src/content/tutorials/nbm-java-hint.asciidoc

```diff
@@ -29,27 +29,25 @@
 :experimental:
 :description: NetBeans Java Hint Module Tutorial - Apache NetBeans
 :keywords: Apache NetBeans Platform, Platform Tutorials, NetBeans Java Hint Module Tutorial
+:reviewed: 2022-02-07
 
 This tutorial demonstrates how to create a NetBeans module that provides one or more Java hints. At the end of the first scenario covered in this tutorial, whenever the user types " ``JOptionPane.showMessageDialog`` ", a Java hint will appear, asking the user whether the statement is needed (or whether it is there because the user needed it for debugging):
 
 At the end of the second scenario, the user will be able to click the hint to delete the superfluous statement:
 
 In addition, the user will be able to find and replace multiple instances of the superfluous statement, after using the "Inspect &amp; Transform" dialog in the Refactor menu, which displays all found instances in the Refactoring window:
 
-NOTE:  This document is applicable to NetBeans IDE 8.0 and NetBeans Platform 8.0. If you are using an earlier version, see  link:74/nbm-java-hint.html[the previous version of this document].
-
 For troubleshooting purposes, you are welcome to download the  link:http://web.archive.org/web/20170409072842/http://java.net/projects/nb-api-samples/show/versions/8.0/tutorials/MyCustomHints[completed tutorial source code].
 
 == Learning About Java Hints
 
 The best way to learn about the Java hint infrastructure is to read the sources of the Java hints that are part of the IDE. Start by identifying a hint that does something similar to what you would like to achieve, then check out the NetBeans sources, and read the source code of the related hint.
 
 [start=1]
-1. Use Mercurial to clone the NetBeans sources:  link:https://netbeans.apache.org/wiki/devfaqaccesssourcesusingmercurial[http://wiki.netbeans.org/DevFaqAccessSourcesUsingMercurial]
+1. Use Mercurial to clone the NetBeans sources:  link:https://netbeans.apache.org/wiki/DevFaqAccessSourcesUsingMercurial[http://wiki.netbeans.org/DevFaqAccessSourcesUsingMercurial]
```

Comment:
should be updated to something like "Use git to clone the NetBeans project: link to github"